### PR TITLE
Move ol-util to typescript

### PR DESCRIPTION
**Pull request by ahennr**
By this PR the code of `ol-util` is moved to typescript

* Basic typings for Utils and Tests are included and exported
* Utils have been completely overhauled ➡️ *breaking change*
* notable changes:
  * introduce types for CrsMapping and CrsDefinition in `ProjectionUtil` (breaking)
  * introduce types for `InkmapLayer`
  * use `terrestris/eslint-config-typescript`
  * remove `getInteractionsByClass` in `MapUtils`

Plz review @terrestris/devs

### CODE REVIEW

**Review by marcjansen — COMMENTED** [review]
Can we please have a relese notes file that tells people how to properly react to the breaking changes?

File: src/FeatureUtil/FeatureUtil.spec.ts [hunk 1]
```diff
@@ -123,10 +122,12 @@ describe('FeatureUtil', () => {
         const mockFn = jest.fn(() => {return 'FOO';});
         got = FeatureUtil.resolveAttributeTemplate(feat, template, '', mockFn);
         expect(mockFn.mock.calls.length).toBe(2);
-        expect(mockFn.mock.calls[0][0]).toBe('exists-and-is-undefined');
-        expect(mockFn.mock.calls[0][1]).toBe(undefined);
-        expect(mockFn.mock.calls[1][0]).toBe('exists-and-is-null');
-        expect(mockFn.mock.calls[1][1]).toBe(null);
+
+        // TODO !
```
Comment (KaiVolland) [inline src/FeatureUtil/FeatureUtil.spec.ts:126]: Will this TODO be solved within this PR or later?

File: tsconfig.json [hunk 2]
```diff
@@ -1,16 +1,20 @@
 {
   "compilerOptions": {
-    "allowJs": true,
-    "checkJs": true,
     "target": "es5",
+    "lib": ["es5", "es6", "dom"],
```
Comment (KaiVolland) [inline tsconfig.json:4]: "esNext" woul be cool

**Review by KaiVolland — APPROVED** [review]
Very nice.

I found some things which could be improved, but that actually don't belong to the topic of your PR. I also addes two minor comments.

**Review by hwbllmnn — APPROVED** [review]

**Comment by ahennr** [comment]
> Can we please have a relese notes file that tells people how to properly react to the breaking changes?

I've added a changelog containing the most significant changes in a098dea615a7d2a18d2291d99de4b4864c7bae49

**Review by marcjansen — APPROVED** [review]
Thanks for picking up my suggestion, and thanks for the migration. Please merge.

**Pull request merged**